### PR TITLE
Decrease effectiveness of timewarp attack

### PR DESCRIPTION
**Pull request by egyptianbman**
This decreases the window of time that a block's timestamp can be in the future from two hours to 30 minutes. I think 30 minutes is still too much -- we can probably get away with 15 minutes but we'll take it one step at a time.

### CODE REVIEW

**Comment by egyptianbman** [comment]
Unfortunately, there doesn't seem to be any interest in fixing this bug.

File: src/main.cpp [hunk 1]
```diff
@@ -3002,8 +3002,16 @@ bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool f
...
-    if (block.GetBlockTime() > GetAdjustedTime() + 2 * 60 * 60)
+    unsigned int nHeight = chainActive.Height();
+    const CChainParams& chainParams = Params();
+
+    // Check timestamp (old)
+    if (nHeight < chainParams.GetNewTimeRule() && block.GetBlockTime() > GetAdjustedTime() + 2 * 60 * 60)
+        return state.Invalid(error("CheckBlockHeader(): block timestamp too far in the future"),
+                             REJECT_INVALID, "time-too-new");
+
+    // starting at height 158750, decrease to 30 minute window to decrease effectiveness of timewarp attack.
```
Comment (cryptorex) [inline src/main.cpp]: change comment to 159,300

File: src/chainparams.cpp [hunk 2]
```diff
@@ -61,6 +61,7 @@ class CMainParams : public CChainParams {
         nDefaultPort = 1989;
         nMaxTipAge = 24 * 60 * 60;
         nPruneAfterHeight = 100000;
+        newTimeRule = 158750;
```
Comment (cryptorex) [inline src/chainparams.cpp]: needs 159,300

**Review by cryptorex — DISMISSED** [review]
ship ship!

File: src/clientversion.h [hunk 3]
```diff
@@ -19,7 +19,7 @@
 #define CLIENT_VERSION_MAJOR 1
 #define CLIENT_VERSION_MINOR 3
```
Comment (cryptorex) [inline src/clientversion.h]: needs bump to 4?

**Review by NickRHill — APPROVED** [review]
Approve

**Review by cryptorex — APPROVED** [review]

**Pull request merged**